app/controllers/foods_controller.rb

```diff
@@ -42,44 +42,61 @@ def update
 
   def destroy
     food = Food.find(params[:id])
-    if @current_user.role === "admin" && food
+    if food
       food.destroy
       flash[:success] = "#{food.name} has been deleted."
-      redirect_to admin_foods_path
+      redirect_to dashboard_path
     else
       flash[:error] = "An error occured. Try deleting #{@food.name} again."
     end
   end
 
   def edit_status
-   if @current_user.role === "admin"
-     food = Food.find(food_params[:id].to_i)
-     if (!food.nil?)
-       status = food_params[:status] == "true" ? "available" : "not available"
-       food.status = status
-       food.save
-       render json: food
-     end
-   end
+    food = Food.find(food_params[:id].to_i)
+    unless food.nil?
+      status = food_params[:status] == "true" ? "available" : "not available"
+      food.status = status
+      food.save
+      render json: food
+    end
   end
 
   private
 
   def food_params
-    params.require(:food).permit(:id, :name, :description, :price, :category_id, :food_image, :status)
+    params.require(:food).permit(:id, :name, :description, :price,
+                                 :category_id, :food_image, :status,
+                                 sales: [:percentage, :price,
+                                 :status, :js])
   end
 
   def upload_image
     image = food_params[:food_image]
     @food_image_url = false
-     if image && image.size < 1.megabytes
-        @upload = {}
-        @upload[:food_image] = Cloudinary::Uploader.upload(image)
-        @food_image_url = @upload[:food_image]["url"]
-        @food[:food_image_file_name] = @food_image_url
-     else
-       flash[:warning] = "file size must be between 0 and 1 megabytes"
-     end
-     @food_image_url
+    if image && image.size < 1.megabytes
+      @upload = {}
+      @upload[:food_image] = Cloudinary::Uploader.upload(image)
+      @food_image_url = @upload[:food_image]["url"]
+      @food[:food_image_file_name] = @food_image_url
+    else
+      flash[:warning] = "file size must be between 0 and 1 megabytes"
+    end
+    @food_image_url
+  end
+
+  def save_food
+    @food[:food_image_file_name] = @food_image_url
+    if food_params[:sales][:js]
+      food = food_params[:sales]
+      food_status = (food["status"] == "true") ? true : false
+      @food.update(sales: { price: food["price"].to_f, 
+                          percentage: food["percentage"].to_f, 
```

Comment:
Align the elements of a hash literal if they span more than one line.<br>Trailing whitespace detected.